File: app/src/main/java/com/phlogiston/gauges/ArcGauge.java
package com.phlogiston.gauges;

import android.content.Context;
import android.graphics.Color;
import android.graphics.Paint;
import android.util.AttributeSet;

public class ArcGauge extends FullGauge {

    private float sweepAngle = 240;
    private float startAngle = 150;
    private float gaugeBGWidth = 20f;

    public ArcGauge(Context context) {
        super(context);
        init();
    }

    public ArcGauge(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public ArcGauge(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    public ArcGauge(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init();
    }

    private void init() {
        getGaugeBackGround().setStrokeWidth(gaugeBGWidth);
        getGaugeBackGround().setStrokeCap(Paint.Cap.ROUND);
        getGaugeBackGround().setColor(Color.parseColor("#D6D6D6"));
        getTextPaint().setTextSize(35f);
        setPadding(20f);
        setSweepAngle(sweepAngle);
        setStartAngle(startAngle);
    }
}

File: app/src/main/java/com/phlogiston/gauges/ArcGauge.kt
package com.phlogiston.gauges

import android.content.Context
import android.graphics.Color
import android.os.Build
import android.support.annotation.RequiresApi
import android.util.AttributeSet

internal class ArcGauge : FullGauge {

    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init()
    }

    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
        init()
    }

    private fun init() {
        getGaugeBackground().strokeWidth = 50f
        getGaugeBackground().color = Color.parseColor("#D6D6D6")
        //getGaugeBackground().strokeCap = Paint.Cap.ROUND
        getTextPaint().textSize = 35f
        padding = 45f
        startAngle = 157.5f
        sweepAngle = 225f
        isDrawSectorLines = true
    }
}
